src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp

```diff
@@ -187,26 +188,58 @@ static int removeDevice(unsigned int index, bool is_userpf)
         return -EINVAL;
     }
 
+    /* Poll till remove is done */
+    do {
+        sleep(1);
+        std::string path = dev->get_sysfs_path("", "remove");
+        if (!boost::filesystem::exists(path))
+            break;
+
+    } while (++wait < POLL_TIMEOUT );
+
```

Comment:
Why polling remove? Is echo 1 >/sys/.../remove a sync call?

Reply:
In some setup ever after echo 1 > remove comeout successfully,, the sysfs files are taking some time to disappear. and xbmgmt scan is still showing the all the xilinx devices. Hence some of the test scripts are failing. Please check the below output:

[root@xsjdsv08 xbmgmt]# ./xbmgmt  scan
 0000:89:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=35072)
 0000:3e:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=15872)
[root@xsjdsv08 xbmgmt]# ls /sys/bus/pci/devices/0000\:89\:00.0/mgmt_pf 
/sys/bus/pci/devices/0000:89:00.0/mgmt_pf
[root@xsjdsv08 xbmgmt]# ./xbmgmt hotplug --offline 0000:89:00.0
CAUTION: Performing hotplug command. This command is going to impact both user pf and mgmt pf.
Please make sure no application is currently running.
Are you sure you wish to proceed? [y/n]: y
[root@xsjdsv08 xbmgmt]# ls /sys/bus/pci/devices/0000\:89\:00.0/mgmt_pf 
/sys/bus/pci/devices/0000:89:00.0/mgmt_pf
[root@xsjdsv08 xbmgmt]# ./xbmgmt  scan
 0000:89:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=35072)
 0000:3e:00.0 xilinx_samsung_u2x4_201920_2(ID=0x5e3d30a2) mgmt(inst=15872)
[root@xsjdsv08 xbmgmt]# ls /sys/bus/pci/devices/0000\:89\:00.0/mgmt_pf 
ls: cannot access /sys/bus/pci/devices/0000:89:00.0/mgmt_pf: No such file or directory
[root@xsjdsv08 xbmgmt]#

Reply:
In the hotplug case, please echo 1 > <parent>/remove. That should be a sync call and you should not see both userpf and mgmtpf node after echo 1 returned.